plugin/aws/s3Object.go

```diff
@@ -23,16 +22,38 @@ type s3Object struct {
 	client *s3Client.S3
 }
 
-func newS3Object(name string, bucket string, key string, client *s3Client.S3) *s3Object {
-	o := &s3Object{
+func newS3Object(o *s3Client.Object, name string, bucket string, key string, client *s3Client.S3) *s3Object {
+	s3Obj := &s3Object{
 		EntryBase: plugin.NewEntry(name),
 		bucket:    bucket,
 		key:       key,
 		client:    client,
 	}
-	o.DisableCachingFor(plugin.Metadata)
-
-	return o
+	s3Obj.DisableCachingFor(plugin.MetadataOp)
+
+	// S3 objects do not have a "creation time"; they're treated as atomic
+	// blobs that get replaced whenever the user uploads new data. Thus, we
+	// use the mtime as its creation date. See https://stackoverflow.com/questions/27746760/how-do-i-get-the-s3-keys-created-date-with-boto
+	// for more details.
+	//
+	// TODO: Export a mungeSize helper to abstract away the common
+	// logic of validating a negative size
+	mtime := awsSDK.TimeValue(o.LastModified)
+	attr := plugin.EntryAttributes{}
+	attr.
+		SetCtime(mtime).
+		SetMtime(mtime).
+		SetAtime(mtime).
+		SetSize(uint64(awsSDK.Int64Value(o.Size)))
```

Comment:
Should this also include `SetMeta`?

Reply:
No, b/c [head-object](https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#S3.HeadObject) returns something different than what [list-objects returns](https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#ListObjectsOutput).

This is brings up an important question -- do we want to maybe define `meta` to represent the complete metadata returned by a `List` endpoint? For example, I noticed that List for containers returns a decent amount of info (though not at the level of detail that RawInspect returns). That would make `find` more powerful and performant -- maybe we can add an option to use the detailed `meta` ?